pkg/webhook/controlplane/ensurer.go

```diff
@@ -357,92 +284,42 @@ var (
-	if csiEnabled && csiMigrationComplete {
-		ps.Volumes = extensionswebhook.EnsureNoVolumeWithName(ps.Volumes, cloudProviderConfigVolume.Name)
-		ps.Volumes = extensionswebhook.EnsureNoVolumeWithName(ps.Volumes, cloudProviderSecretVolume.Name)
-		return
-	}
-
-	ps.Volumes = extensionswebhook.EnsureVolumeWithName(ps.Volumes, cloudProviderConfigVolume)
-	ps.Volumes = extensionswebhook.EnsureVolumeWithName(ps.Volumes, cloudProviderSecretVolume)
+func ensureKubeAPIServerVolumes(ps *corev1.PodSpec) {
+	ps.Volumes = extensionswebhook.EnsureNoVolumeWithName(ps.Volumes, cloudProviderConfigVolume.Name)
+	ps.Volumes = extensionswebhook.EnsureNoVolumeWithName(ps.Volumes, cloudProviderSecretVolume.Name)
 }
 
-func ensureKubeControllerManagerVolumes(ps *corev1.PodSpec, version string, csiEnabled, csiMigrationComplete bool) {
-	if csiEnabled && csiMigrationComplete {
-		ps.Volumes = extensionswebhook.EnsureNoVolumeWithName(ps.Volumes, cloudProviderConfigVolume.Name)
-		ps.Volumes = extensionswebhook.EnsureNoVolumeWithName(ps.Volumes, cloudProviderSecretVolume.Name)
-		ps.Volumes = extensionswebhook.EnsureNoVolumeWithName(ps.Volumes, etcSSLVolume.Name)
-		ps.Volumes = extensionswebhook.EnsureNoVolumeWithName(ps.Volumes, usrShareCaCertsVolume.Name)
-		return
-	}
-
-	ps.Volumes = extensionswebhook.EnsureVolumeWithName(ps.Volumes, cloudProviderConfigVolume)
-	ps.Volumes = extensionswebhook.EnsureVolumeWithName(ps.Volumes, cloudProviderSecretVolume)
-
-	ps.Volumes = extensionswebhook.EnsureVolumeWithName(ps.Volumes, etcSSLVolume)
-	// some distros have symlinks from /etc/ssl/certs to /usr/share/ca-certificates
-	ps.Volumes = extensionswebhook.EnsureVolumeWithName(ps.Volumes, usrShareCaCertsVolume)
+func ensureKubeControllerManagerVolumes(ps *corev1.PodSpec) {
+	ps.Volumes = extensionswebhook.EnsureNoVolumeWithName(ps.Volumes, cloudProviderConfigVolume.Name)
+	ps.Volumes = extensionswebhook.EnsureNoVolumeWithName(ps.Volumes, cloudProviderSecretVolume.Name)
+	ps.Volumes = extensionswebhook.EnsureNoVolumeWithName(ps.Volumes, etcSSLVolume.Name)
+	ps.Volumes = extensionswebhook.EnsureNoVolumeWithName(ps.Volumes, usrShareCaCertsVolume.Name)
 }
 
-func (e *ensurer) ensureChecksumAnnotations(ctx context.Context, template *corev1.PodTemplateSpec, namespace string, csiEnabled, csiMigrationComplete bool) error {
-	if csiEnabled && csiMigrationComplete {
-		if template.Annotations != nil {
-			delete(template.Annotations, "checksum/secret-"+v1beta1constants.SecretNameCloudProvider)
-			delete(template.Annotations, "checksum/configmap-"+internal.CloudProviderConfigName)
-		}
-		return nil
+func (e *ensurer) ensureChecksumAnnotationsDeleted(template *corev1.PodTemplateSpec) {
+	if template.Annotations != nil {
+		delete(template.Annotations, "checksum/secret-"+v1beta1constants.SecretNameCloudProvider)
+		delete(template.Annotations, "checksum/configmap-"+internal.CloudProviderConfigName)
```

Comment:
This handling also could be removed when we drop the `EnsureNoVolumeMountWithName` handling for the cloud-provider-config ConfigMap.